File: solutions/go/two-fer/1/two_fer.go
// Package twofer implements a function to determine what you will say
// as you give away the extra cookie.
package twofer

// ShareWith receives a string as argument representing a person name
// and return a formatted string
func ShareWith(name string) string {
    if name != "" {
        return "One for " + name + ", one for me."
    }
    return "One for you, one for me."
}

File: solutions/erlang/two-fer/1/src/two_fer.erl
-module(two_fer).

-export([two_fer/0, two_fer/1]).

two_fer() -> %% undefined.
  "One for you, one for me.".

two_fer(_Name) -> %% undefined.
    io_lib:format("One for ~s, one for me.", [_Name]).
